code/modules/projectiles/guns/energy/pulse.dm

```diff
@@ -10,6 +10,7 @@
 	charge_cost=100
 	max_shots = 20 // This is cut in half by "DESTROY" mode.
 	sel_mode = 2
+	w_class = 4

```

Comment:
I believe so? All weapons can melt walls. I'm not sure if the pulse rifle has any special code for it. I just bumped it up to 4 because it's a rifle like the other rifles. I don't really see any need to make it 5 and balance it, it's not really a commonly usable thing. The people who do have it, probably should just have it as 4 due to the circumstances.